sdk_extensions/logging/support/src/main/java/io/opentelemetry/logging/api/LogRecord.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.logging.api;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.common.AnyValue;
+import io.opentelemetry.common.AttributeValue;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * A LogRecord is an implementation of the <a
+ * href="https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/logs/data-model.md">
+ * OpenTelemetry logging model</a>.
+ */
+@AutoValue
+public abstract class LogRecord {
+  abstract long getTimeUnixNano();
+
+  @SuppressWarnings("mutable")
+  abstract byte[] getTraceId();
```

Comment:
SpanData uses string instead of byte[]: https://github.com/open-telemetry/opentelemetry-java/blob/d138908996af46e3fa38f6cb82820b10b72e8752/sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/data/SpanData.java#L46

I don't know what is better in Java but let's be consistent.

Reply:
we avoided using `byte[]` specifically because it's a mutable type, where as String is immutable.